TFNetworkRecLayer.py

```diff
@@ -3106,10 +3168,98 @@ def __init__(self, beam_size, input_type="prob", explicit_search_source=None, le
       assert len(self.sources) == 0  # will be filtered out in transform_config_dict
       # Note: If you want to do forwarding, without having the reference,
       # that wont work. You must do search in that case.
-      self.output = self._static_get_target_value(
-        target=self.target, network=self.network,
-        mark_data_key_as_used=True).copy()
-      self.output.available_for_inference = True  # in inference, we should do search
+      targets = self.target
+      if isinstance(targets, str):
+        targets = [targets]
+      # Put all targets in a list.
+      # They can be accessed by using the sublayers created in self.get_sub_layer().
+      self.output_list = []
+      for target in targets:
+        target_out_data = self._static_get_target_value(
+            target=target, network=self.network, mark_data_key_as_used=True).copy()
+        target_out_data.available_for_inference = True  # in inference, we should do search
+        self.output_list.append(target_out_data)
+
+      # We use the labels of the first target as "normal" output.
+      self.output = self.output_list[0]
+
+  def _get_scores(self, source):
+    scores_in = source.output.placeholder
+    # We present the scores in +log space, and we will add them up along the path.
+    if self.input_type == "prob":
+      if source.output_before_activation:
+        return source.output_before_activation.get_log_output()
+      else:
+        from TFUtil import safe_log
+        return safe_log(scores_in)
+    elif self.input_type == "log_prob":
+      return scores_in
+    else:
+      raise Exception("%r: invalid input type %r" % (self, self.input_type))
+
+  def _prune_and_combine_sources(self, sources, beam_sizes, batch_dim):
+    """
+    Applies beam bruning to the sources and then calculates all possible sums of scores.
+    Returns the scores, the (static) number of targets after pruning and a list of
+    labels corresponding to the top scores.
+    :param list[LayerBase] sources: input layers providing the scores
+    :param list[int] beam_sizes: beam sizes used for pruning of the individual sources
+    :param tf.Tensor|int batch_dim: dim of batch axis (batch size * incoming beam)
+    :rtype: (tf.Tensor, int, list[tf.Tensor])
+    """
```

Comment:
Documentation missing (`:return: ...`). Just reformat your text "Returns the scores ...".

Reply:
Btw, not sure about this, but I think readthedocs/Sphinx has problems when there is no empty line between your description and the first parameter (`:param ...`). Maybe check that and if needed, add an empty line in between.